src/types/message.ts

```diff
@@ -0,0 +1,47 @@
+export enum MessageType {
+  Result = "Result",
+  Raw = "Raw"
+}
+
+export enum MessageLevel {
+  Info = "Info",
```

Comment:
You used string enum here. How about union type? 

`MessageLevel = "Info"|"Error"`

Reply:
yes, agreed with you, updated as unions